### ISSUE TITLE
Facts are not set properly

### ISSUE THREAD

**Issue by Kriechi** [invadersmustdie/puppet-catalog-test#24]
I'm using a module which checks `osfamily` and fails if it is not `Debian`.
Now I tried to test it with a scenario for one node, but everytime the module complains that `osfamily` is empty!.

I checked and tried debugging the code, and can see that my `scenarios.yml` is read properly and all the facts show up here (https://github.com/invadersmustdie/puppet-catalog-test/blob/master/lib/puppet-catalog-test/test_runner.rb#L77) where it is passed of to the Puppet API. After that I'm not really sure what happens.

Is this a bug, or did I miss some configuration step?

My `scenarios.yml`:
```yaml
__default_facts: &default_facts
  architecture: x86_64
  ipaddress: 127.0.0.1
  osfamily: Debian
  operatingsystem: Ubuntu
  operatingsystemrelease: 15.10
  local_run: true
  disable_asserts: true
  interfaces: eth0

foobar:
  <<: *default_facts
  fqdn: foobar.example.com
```

Specifically I'm trying to use puppetlabs-rabbitmq, and it fails on this check: https://github.com/puppetlabs/puppetlabs-rabbitmq/blob/master/manifests/params.pp#L62
with the following error message:
```
Evaluation Error: Error while evaluating a Function Call, The rabbitmq module is not supported on an  based system. at <...>/modules-contrib/rabbitmq/manifests/params.pp:62:7 on node foobar
```
As you can see, the part says: `... on an  based system` - clearly the actual `osfamily` variable is empty.

**Comment by jerith** [comment]
The problem here is that the facts are only available through the `$facts` toplevel variable. They also need to be merged into the node's scope, as in the 3.x case. The following change to `PuppetCatalogTest::Puppet4xAdapter#create_node` fixes the issue:

```ruby
def create_node(hostname, facts)
  node = Puppet::Node.new(hostname, :facts => Puppet::Node::Facts.new(hostname, facts))
  node.merge(facts)
  node
end
```

**Comment by invadersmustdie** [comment]
This should be fixed in next release, which is scheduled for this weekend. Thanks @jerith for supplying the fix!

**Comment by invadersmustdie** [comment]
just released 0.4.2
